blobfile/_azure.py

```diff
@@ -1212,14 +1219,15 @@ def _upload_chunk(self, chunk: memoryview, finalize: bool) -> None:
             block_ids = [
                 _block_index_to_block_id(i, self._upload_id) for i in range(self._block_index)
             ]
-            _finalize_blob(
+            resp = _finalize_blob(
                 conf=self._conf,
                 path=self._path,
                 url=self._url,
                 block_ids=block_ids,
                 md5_digest=self._md5.digest(),
                 version=self._version,
             )
+            self._version = resp.headers.get("ETag") or self._version

```

Comment:
IIUC this isn't needed? The file is already finalized right?
I also kind of liked the other order better: `self._version = self._version or resp.headers.get("ETag")`

Reply:
Hm, not sure I follow. This is the ETag returned by the "put blocklist" operation. It should be different from the previously seen ETag, and it represents the version of the file after the write completes. I want `self._version` to reflect the ETag of the last version of the file seen, so I should update it here, and the new result should take precedence.